scipy/io/tests/test_wavfile.py

```diff
@@ -84,11 +84,23 @@ def test_read_fail():
         fp.close()
 
 
+def test_read_early_eof_with_data():
+    for mmap in [False, True]:
+        with open(datafile('test-44100Hz-le-1ch-4bytes-early-eof.wav'), 'rb') as fp:
+            with suppress_warnings() as sup:
+                sup.filter(wavfile.WavFileWarning,
+                            r"Reached EOF .filesize reported as \d+ by header.\."
+                           )
+                rate, data = wavfile.read(fp, mmap=mmap)
```

Comment:
A better way to do it would be:
```
with pytest.warns(wavfile.WavFileWarning, match='Reached EOF'):
    rate, data = wavefile.read(fp, mmap=mmap)
    assert data.size > 0
    assert rate = ...   # whatever it's actually supposed to be
```

Reply:
I loaded in `pytest.warns` as `assert_warns` to mirror `import raises as assert_raises`.